### PR TITLE
Fixes #37464 - Support Zeitwerk loader

### PR DESCRIPTION
**Pull request by ofedoren**
~WIP~

Requires:
 - https://github.com/theforeman/foreman-tasks/pull/752

### CODE REVIEW

File: lib/foreman_openscap/engine.rb [hunk 1]
```diff
@@ -7,13 +7,6 @@ class Engine < ::Rails::Engine
     engine_name 'foreman_openscap'
     config.autoload_paths += Dir["#{config.root}/app/controllers/concerns"]
     config.autoload_paths += Dir["#{config.root}/app/helpers/concerns"]
```
Comment (ekohl) [inline lib/foreman_openscap/engine.rb:9]: Why do these 2 remain, but you remove the others?

Comment (ofedoren) [reply lib/foreman_openscap/engine.rb]: I think there is no reason to leave them, removed.

**Comment by ofedoren** [comment]
Hmm, not sure why the tests from core are failing here... I had the same errors for plain Foreman, but they got fixed.

**Comment by ekohl** [comment]
I think this may be a relevant failure, coming from STI with interfaces.

**Comment by ofedoren** [comment]
@ekohl thanks, it was indeed. Moreover, thanks to that fail we prevented a hidden issue in REX caused by the same issue.

I'm afraid there might be still places we didn't catch by the test runs :/

**Comment by ofedoren** [comment]
Ready to merge after https://github.com/theforeman/foreman/pull/10131 is merged, dropped the commit for testing with Foreman PR, the latest run was green:
Screenshot_20240903_143351

**Comment by ofedoren** [comment]
tasks and REX are done, ansible is pending

**Review by ekohl — APPROVED** [review]
Note to self: foreman_ansible is a soft dependency so there's no version check that needs to be updated.

**Pull request merged**